### ISSUE TITLE
Validation errror message: string gets expanded to list of characters

### ISSUE THREAD

**Issue by evgeny-roubinchtein** [aws/serverless-application-model#2325]
<!-- Make sure we don't have an existing Issue that reports the bug you are seeing (both open and closed). 
If you do find an existing Issue, re-open or add a comment to that Issue instead of creating a new one. -->

### Description:
In an error message, string gets expanded to a list of characters, making it hard to read.  Can you read the following?:
```
2022-02-15 11:43:29 Template schema validation reported the following errors: [, P, a, r, a, m, e, t, e, r, s, ., L, a, t, e, s, t, A, m, i, I, d, ., T, y, p, e, ],  , ', A, W, S, :, :, S, S, M, :, :, P, a, r, a, m, e, t, e, r, :, :, V, a, l, u, e, <, A, W, S, :, :, E, C, 2, :, :, I, m, a, g, e, :, :, I, d, >, ',  , i, s,  , n, o, t,  , o, n, e,  , o, f,  , [, ', S, t, r, i, n, g, ', ,,  , ', N, u, m, b, e, r, ', ,,  , ', L, i, s, t, <, N, u, m, b, e, r, >, ', ,,  , ', C, o, m, m, a, D, e, l, i, m, i, t, e, d, L, i, s, t, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, A, v, a, i, l, a, b, i, l, i, t, y, Z, o, n, e, :, :, N, a, m, e, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, I, m, a, g, e, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, I, n, s, t, a, n, c, e, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, K, e, y, P, a, i, r, :, :, K, e, y, N, a, m, e, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, G, r, o, u, p, N, a, m, e, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, S, u, b, n, e, t, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, V, o, l, u, m, e, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, V, P, C, :, :, I, d, ', ,,  , ', A, W, S, :, :, R, o, u, t, e, 5, 3, :, :, H, o, s, t, e, d, Z, o, n, e, :, :, I, d, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, A, v, a, i, l, a, b, i, l, i, t, y, Z, o, n, e, :, :, N, a, m, e, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, I, m, a, g, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, I, n, s, t, a, n, c, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, G, r, o, u, p, N, a, m, e, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, S, u, b, n, e, t, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, V, o, l, u, m, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, V, P, C, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, R, o, u, t, e, 5, 3, :, :, H, o, s, t, e, d, Z, o, n, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, S, t, r, i, n, g, >, ', ]
```



### Steps to reproduce:
<!-- Provide detailed steps to replicate the bug, including steps from third party tools (CDK, etc.) -->

Run `sam validate -t mappings.yaml`, where `mappings.yaml` is the following slightly modified file from the [The AWS CloudFormation Workshop](https://cfn101.workshop.aws/basics/templates/mappings.html):
```AWSTemplateFormatVersion: "2010-09-09"                                                                                                                                                                               
Transform: AWS::Serverless-2016-10-31                                                                                                                                                                                
                                                                                                                                                                                                                     
Description: AWS CloudFormation workshop - Mappings (uksb-1q9p31idr).                                                                                                                                                
                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
Parameters:                                                                                                                                                                                                          
  EnvironmentType:                                                                                                                                                                                                   
    Description: 'The environment (Test or Prod) in which the stack runs.'                                                                                                                                           
    Type: String                                                                                                                                                                                                     
    Default: Test                                                                                                                                                                                                    
    AllowedValues:                                                                                                                                                                                                   
      - Test                                                                                                                                                                                                         
      - Prod                                                                                                                                                                                                         
    ConstraintDescription: 'Either "Test" or "Prod"'                                                                                                                                                                 
                                                                                                                                                                                                                     
  LatestAmiId:                                                                                                                                                                                                       
    Type: 'AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>'                                                                                                                                                          
    Default: '/aws/service/ami-amazon-linux-latest/amzn2-ami-kernel-5.10-hvm-x86_64-ebs'                                                                                                                             
                                                                                                                                                                                                                     
Mappings:                                                                                                                                                                                                            
  EnvironmentToInstanceType:                                                                                                                                                                                         
    Test:                                                                                                                                                                                                            
      InstanceType: t2.micro                                                                                                                                                                                         
    Prod:                                                                                                                                                                                                            
      InstanceType: t2.small                                                                                                                                                                                         
                                                                                                                                                                                                                     
Resources:                                                                                                                                                                                                           
  WebServerInstance:                                                                                                                                                                                                 
    Type: AWS::EC2::Instance                                                                                                                                                                                         
    Properties:                                                                                                                                                                                                      
      ImageId: !Ref LatestAmiId                                                                                                                                                                                      
      InstanceType: !FindInMap                                                                                                                                                                                       
        - EnvironmentToInstanceType                                                                                                                                                                                  
        - !Ref EnvironmentType                                                                                                                                                                                       
        - InstanceType                                                                                                                                                                                               
      Tags:                                                                                                                                                                                                          
        - Key: Name                                                                                                                                                                                                  
          Value: !Join [ '-', [ !Ref EnvironmentType, webserver ] ]      
```



### Observed result:
<!-- Please provide command output with `--debug` flag set.-->
You get this in your terminal:
```
$ sam validate -t mappings.yaml --region us-west-2
2022-02-15 11:43:28 Loading policies from IAM...
2022-02-15 11:43:29 Finished loading policies from IAM.
2022-02-15 11:43:29 Template schema validation reported the following errors: [, P, a, r, a, m, e, t, e, r, s, ., L, a, t, e, s, t, A, m, i, I, d, ., T, y, p, e, ],  , ', A, W, S, :, :, S, S, M, :, :, P, a, r, a, m, e, t, e, r, :, :, V, a, l, u, e, <, A, W, S, :, :, E, C, 2, :, :, I, m, a, g, e, :, :, I, d, >, ',  , i, s,  , n, o, t,  , o, n, e,  , o, f,  , [, ', S, t, r, i, n, g, ', ,,  , ', N, u, m, b, e, r, ', ,,  , ', L, i, s, t, <, N, u, m, b, e, r, >, ', ,,  , ', C, o, m, m, a, D, e, l, i, m, i, t, e, d, L, i, s, t, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, A, v, a, i, l, a, b, i, l, i, t, y, Z, o, n, e, :, :, N, a, m, e, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, I, m, a, g, e, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, I, n, s, t, a, n, c, e, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, K, e, y, P, a, i, r, :, :, K, e, y, N, a, m, e, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, G, r, o, u, p, N, a, m, e, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, S, u, b, n, e, t, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, V, o, l, u, m, e, :, :, I, d, ', ,,  , ', A, W, S, :, :, E, C, 2, :, :, V, P, C, :, :, I, d, ', ,,  , ', A, W, S, :, :, R, o, u, t, e, 5, 3, :, :, H, o, s, t, e, d, Z, o, n, e, :, :, I, d, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, A, v, a, i, l, a, b, i, l, i, t, y, Z, o, n, e, :, :, N, a, m, e, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, I, m, a, g, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, I, n, s, t, a, n, c, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, G, r, o, u, p, N, a, m, e, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, S, e, c, u, r, i, t, y, G, r, o, u, p, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, S, u, b, n, e, t, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, V, o, l, u, m, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, E, C, 2, :, :, V, P, C, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, A, W, S, :, :, R, o, u, t, e, 5, 3, :, :, H, o, s, t, e, d, Z, o, n, e, :, :, I, d, >, ', ,,  , ', L, i, s, t, <, S, t, r, i, n, g, >, ', ]
/home/ec2-user/src/cfn101-workshop/code/workspace/mappings.yaml is a valid SAM Template
2022-02-15 11:43:30 There are unpublished metrics. Please make sure you call publish after you record all metrics.
2022-02-15 11:43:30 There are unpublished metrics. Please make sure you call publish after you record all metrics.
```



### Expected result:
<!-- Describe what you expected.-->

For the error message to be printed as a string, and not be expanded into a list of characters.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Linux
2. If using SAM CLI, `sam --version`: SAM CLI, version 1.37.0
3. AWS region: us-west-2

`Add --debug flag to any SAM CLI commands you are running` (attached as file `output.txt`)
[output.txt](https://github.com/aws/serverless-application-model/files/8074251/output.txt)

**Comment by moelasmar** [comment]
it seems it is fixed in SAM 
https://github.com/aws/serverless-application-model/blob/979825fdf44b5f06036996f0446b872bf52bc729/samtranslator/parser/parser.py#L71

**Comment by moelasmar** [comment]
it is fixed in this PR #2245 .. waiting to be released

**Comment by jfuss** [comment]
Released in [v1.43.0](https://github.com/aws/serverless-application-model/releases/tag/v1.43.0)

Closing
